activerecord/lib/active_record/tasks/mysql_database_tasks.rb

```diff
@@ -71,7 +74,13 @@ def structure_load(filename)
         args = prepare_command_options('mysql')
         args.concat(['--execute', %{SET FOREIGN_KEY_CHECKS = 0; SOURCE #{filename}; SET FOREIGN_KEY_CHECKS = 1}])
         args.concat(["--database", "#{configuration['database']}"])
-        Kernel.system(*args)
+        
+        case Kernel.system(*args)
+        when nil
+          raise "Unable to load structure, is mysql installed?"
+        when false
+          raise "Error loading structure, mysql exited with status #{$?.exitstatus}"
+        end
```

Comment:
@senny sure we can.  What do you think about modifying run_cmd_error to differentiate command not found from a regular failure?